### ISSUE TITLE
Improving design consistency of fuzzy finder

### ISSUE THREAD

**Issue by jjinnii** [sourcegraph/sourcegraph#43154]
This issue is intended to improve the design consistency and scalability of components used in the newly enabled fuzzy finder:

- [ ] Loading state - Add loading state for when large number of results are being calculated
- [x] List view - Make search results list consistent with Wildcard
- [x] Tabs - Make navigation tabs consistent with Wildcard tabs
- [x] Search everywhere checkbox - Update search everwhere checkbox to Wildcard dropdown form field
- [x] Live updating search query: education on search language (onboarding) 
- [ ] Search results
  - [ ]   ~~Files - Hard to distinguish if a user enables "search everywhere"~~
  - [ ]   Symbols - Metadata for symbols results is hard to scan 

[Figma](https://www.figma.com/file/E0vD9UduSp2liVyDb7dplw/Fuzzy-finder-UI-improvements?node-id=1714%3A3559)

Misalignments added by @felixfbecker:

- [x] The results list displays file names in monospace and link-blue, while in the file tree we use regular color and font ([Figma comment](https://www.figma.com/file/E0vD9UduSp2liVyDb7dplw?node-id=1714:3559#311029195)).
  - @olafurpg mentioned some readability issues with non-monospace but maybe those would not be an issue if the color contrast wasn't blue on yellow.
- [ ] The symbol icons used are different from what we use in the symbols sidebar (I think this may actually be a violation of WCAG [3.2.4: Consistent Identification](https://www.w3.org/WAI/WCAG21/Understanding/consistent-identification.html))

**Comment by jjinnii** [comment]
@almeidapaulooliveira Looping you into the consultative work I'm doing to improve the design consistency of the new fuzzy finder. @rrhyne mentioned that you were potentially taking this on, to incorporate search history into this window. Maybe we can collaborate on this?

**Comment by jjinnii** [comment]
FYI @olafurpg this is the [draft Figma](https://www.figma.com/file/E0vD9UduSp2liVyDb7dplw/Fuzzy-finder-UI-improvements?node-id=1714%3A3559) -- I'll be dropping some questions in there for you shortly!

**Comment by almeidapaulooliveira** [comment]
@jjinnii, cool, thanks for the loop. We started exploring the [search history on the sidebar](https://www.figma.com/file/XAa0yGjuz9k95P0htDhfVE/UX%3A-Search-history?node-id=53%3A2524). For now, I don't think we'll pursue the idea of having it on Fuzzy Finder.

**Comment by felixfbecker** [comment]
I think at this point the only design inconsistencies I can see is:

- The results list displays file names in monospace and link-blue, while in the file tree we use regular color and font ([Figma comment](https://www.figma.com/file/E0vD9UduSp2liVyDb7dplw?node-id=1714:3559#311029195)).
  - @olafurpg mentioned some readability issues with non-monospace but maybe those would not be an issue if the color contrast wasn't blue on yellow.
- The symbol icons used are different from what we use in the symbols sidebar (I think this may actually be a violation of WCAG [3.2.4: Consistent Identification](https://www.w3.org/WAI/WCAG21/Understanding/consistent-identification.html))

**Comment by olafurpg** [comment]
> The symbol icons used are different from what we use in the symbols sidebar (I think this may actually be a violation of WCAG [3.2.4: Consistent Identification](https://www.w3.org/WAI/WCAG21/Understanding/consistent-identification.html))

I'm OK with using the same symbol icons as the symbol sidebar. The current icons are the same as we use for `type:symbol` search results. If these icons are a WCAG violation then we should remove the entirely from the app.

> The results list displays file names in monospace and link-blue, while in the file tree we use regular color and font ([Figma comment](https://www.figma.com/file/E0vD9UduSp2liVyDb7dplw?node-id=1714:3559#311029195)).

Can we update the files sidebar to use monospace font? I tried disabling monospace font in the fuzzy finder and it was a big usability regression.

**Comment by felixfbecker** [comment]
> Can we update the files sidebar to use monospace font? I tried disabling monospace font in the fuzzy finder and it was a big usability regression.

Could you elaborate on the usability regression? GitHub, Gitlab, VS Code all use non-monospace font for their file trees and fuzzy finders.

**Comment by olafurpg** [comment]
I shared a screenshot in https://sourcegraph.slack.com/archives/C03CSAER9LK/p1666885301422609?thread_ts=1666813711.700039&cid=C03CSAER9LK with a side-by-side comparison. It's noticeably harder to interpret the non-monospace results since they're misaligned.

**Comment by olafurpg** [comment]
The screenshots 
![CleanShot 2022-10-27 at 15 20 12](https://user-images.githubusercontent.com/1408093/203271986-d0ea950b-8fea-495b-ab52-680d1b51b806.png)
![CleanShot 2022-10-27 at 15 20 29](https://user-images.githubusercontent.com/1408093/203271992-a3ca06a0-12dd-425d-b3c3-615890f960ab.png)

**Comment by rrhyne** [comment]
cc @quinnkeast

**Comment by felixfbecker** [comment]
@olafurpg do you feel the same way about GitHub's, Gitlab's and VS Code's fuzzy file finders and our file search results?

<img width="780" alt="Screenshot 2022-11-22 at 12 52 27" src="https://user-images.githubusercontent.com/10532611/203425617-6d4a5316-4ba3-431a-b3b6-ff01eeeb323b.png">

<img width="606" alt="Screenshot 2022-11-22 at 13 32 54" src="https://user-images.githubusercontent.com/10532611/203425628-ecbfdb67-e728-4958-99bb-83ba26e6cbd3.png">

<img width="583" alt="image" src="https://user-images.githubusercontent.com/10532611/203425999-22d610c2-c4a0-47d3-969a-4207a116fd84.png">


If I understand you correctly you perceive the same problem with our file tree and potentially in other places too. I think changing it everywhere is a larger change, and for consistency, we should have the same font everywhere for file path results.

I'd suggest that in the interest of getting the fuzzy finder over the finish line, we should make it consistent with file tree/search results/etc, but we could explore separately afterward making monospace font for all file paths a configuration setting, maintaining consistency.

I increasingly think we're never going to satisfy all users with these decisions, and the more important aspect is that we don't make contradictory decisions across different parts of the app and everything starts feeling cobbled together.

**Comment by olafurpg** [comment]
Fine by me to use non-monospace font it that helps enable the fuzzy finder by default.

**Comment by taras-yemets** [comment]
Font family and color feedback is addressed in 
- https://github.com/sourcegraph/sourcegraph/pull/44813

**Comment by taras-yemets** [comment]
Currently we use two different approaches to render symbol icons: [`SymbolIcon`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@f1daa111be3c44b70b0fbfd42ad9c8b1310c0b74/-/blob/client/shared/src/symbols/SymbolIcon.tsx) for [`RepoRevisionSidebarSymbols`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@f1daa11/-/blob/client/web/src/repo/RepoRevisionSidebarSymbols.tsx?L221-233) and [`SymbolTag`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@f1daa11/-/blob/client/shared/src/symbols/SymbolTag.tsx) for the `type:symbol` results page ([`SymbolSearchResult`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@f1daa11/-/blob/client/search-ui/src/components/SymbolSearchResult.tsx?L159)). The latter approach is used by fuzzy finder symbol results tab too ([`FuzzySymbols`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@f1daa11/-/blob/client/web/src/components/fuzzyFinder/FuzzySymbols.tsx?L69)).

| Hierarchical Symbols | Results `type:symbol` | Fuzzy finder symbols tab |
| -- | -- | -- |
| <img width="1028" alt="Screenshot 2022-11-29 at 11 04 31" src="https://user-images.githubusercontent.com/25318659/204486031-26904513-029e-4528-9ff0-6c4d0de9988a.png">|<img width="994" alt="Screenshot 2022-11-29 at 11 05 06" src="https://user-images.githubusercontent.com/25318659/204486043-4ad6d7a3-fd29-4127-913f-07322dabeba7.png">|<img width="991" alt="Screenshot 2022-11-29 at 11 05 23" src="https://user-images.githubusercontent.com/25318659/204486051-77a65fe5-3e1d-4d5e-b32f-333d7c1087bc.png">|

Which approach should we use for the fuzzy finder?
It looks like that whichever approach we choose we'll have follow-up work on either symbols sidebar or `type:symbol` results page. What do you think about addressing the icons inconsistency issue in a follow-up issue?

cc: @felixfbecker, @jjinnii

**Comment by felixfbecker** [comment]
I agree that that is an inconsistency we need to fix too (I think it's actually a WCAG violation), but I feel like the fuzzy finder and symbols sidebar are the most visible, because they are directly next to each other (visible on-screen at the same time) and I believe not that many people use explicit `type:symbol` queries. So I'd still advocate for fixing that inconsistency (fuzzy finder vs symbols sidebar) before making the fuzzy finder the default.

One solution for the overall inconsistency could be to define a component that abstracts over the two implementations based on a feature flag. That way a designer who wants to push for the letter-based icons can gather feedback on them, or a user who prefers them can use that mode.

**Comment by felixfbecker** [comment]
BTW @jjinnii are all the other bullet points listed in your original issue already addressed? Or are there any other outstanding design items you think need to be addressed?

**Comment by jjinnii** [comment]
@felixfbecker Updated the list in this issue so that it's clearer! Only two items remaining, and I don't think they're blockers to making this available externally: 
* Loading state - The result count updates even if the immediate search results (the ones in view) aren't changing. It would be nice to add a loading state to show that results are being calculated. 
* The symbols result path is hard to scan. This is less design consistency with Wildcard and more UX, and I think we can create a new issue for this and wait on more feedback after enabling. WDYT?

Re: symbols @taras-yemets , I agree with Felix that the symbols sidebar is both more heavily used and also more visible (it's visible when fuzzy finder is open) and would opt for making fuzzy finder consistent with hierarchical symbols. It was also my understanding that hierarchical symbols was an improvement that the code intel shipped for 4.0 that we were collecting feedback on, but should be the default for symbols iconography moving forward. Maybe @macraig can provide some insight? If that's the case, we should also open a separate issue to change the type:symbol icons to reflect IMO.

**Comment by taras-yemets** [comment]
@felixfbecker, @jjinnii,
The fuzzy finder symbol icons issue is addressed in https://github.com/sourcegraph/sourcegraph/pull/44940. There is a [follow-up issue](https://github.com/sourcegraph/sourcegraph/issues/44946) for the results `type:symbol` page too.
 
 >Only two items remaining, and I don't think they're blockers to making this available externally:
>- Loading state - The result count updates even if the immediate search results (the ones in view) aren't changing. It would be nice to add a loading state to show that results are being calculated.
>- The symbols result path is hard to scan. This is less design consistency with Wildcard and more UX, and I think we can create a new issue for this and wait on more feedback after enabling. WDYT?

If these are not blockers, I'd move them to separate tickets and close this one when the mentioned PR is merged. If we don't have any other blocking issues we can enable the fuzzy finder by default and gather new and address existing feedback in parallel.
